code/DynamicCache.php

```diff
@@ -397,14 +412,32 @@ public function run($url)
         // This is normally called in VersionedRequestFilter.
         Versioned::choose_site_stage();
 
+        // Get cache
```

Comment:
So I have 1 specific page-type might have say, 40+ variants dependent on a users location (stored in a HTTPOnly cookie), the other pages don't need that many variants (such as the home page).

I don't want 40+ variants of the home page for no good reason so I feel that being able to inspect a page type via cached information helps handle this without hardcoding URL checks.

The reason for not having multiple files anymore per cache is so that when the cache becomes full on our servers, it doesn't end up auto-blowing away say... the meta key for a URL but not the actual cache URL.